src/na/device/Device.cpp

```diff
@@ -0,0 +1,465 @@
+    size_t* sizeRet) {
+  if (session == nullptr || site == nullptr ||
+      (value != nullptr && size == 0) ||
+      (prop != QDMI_SITE_PROPERTY_CUSTOM1 &&
+       prop != QDMI_SITE_PROPERTY_CUSTOM2 &&
+       prop != QDMI_SITE_PROPERTY_CUSTOM3 &&
+       prop != QDMI_SITE_PROPERTY_CUSTOM4 && prop >= QDMI_SITE_PROPERTY_MAX)) {
+    return QDMI_ERROR_INVALIDARGUMENT;
+  }
+  ADD_SINGLE_VALUE_PROPERTY(QDMI_SITE_PROPERTY_INDEX, size_t, site->id, prop,
+                            size, value, sizeRet)
+  ADD_SINGLE_VALUE_PROPERTY(QDMI_SITE_PROPERTY_CUSTOM1, size_t, site->module,
+                            prop, size, value, sizeRet)
+  ADD_SINGLE_VALUE_PROPERTY(QDMI_SITE_PROPERTY_CUSTOM2, size_t, site->subModule,
+                            prop, size, value, sizeRet)
+  ADD_SINGLE_VALUE_PROPERTY(QDMI_SITE_PROPERTY_CUSTOM3, int64_t, site->x, prop,
+                            size, value, sizeRet)
+  ADD_SINGLE_VALUE_PROPERTY(QDMI_SITE_PROPERTY_CUSTOM4, int64_t, site->y, prop,
+                            size, value, sizeRet)
+  ADD_SINGLE_VALUE_PROPERTY(QDMI_SITE_PROPERTY_T1, double,
+                            decoherenceTimes().t1, prop, size, value, sizeRet)
+  ADD_SINGLE_VALUE_PROPERTY(QDMI_SITE_PROPERTY_T2, double,
+                            decoherenceTimes().t2, prop, size, value, sizeRet)
+  return QDMI_ERROR_NOTSUPPORTED;
+}
+
+int MQT_NA_QDMI_device_session_query_operation_property(
+    MQT_NA_QDMI_Device_Session session, MQT_NA_QDMI_Operation operation,
+    const size_t numSites, const MQT_NA_QDMI_Site* sites,
+    const size_t numParams, const double* params,
+    const QDMI_Operation_Property prop, const size_t size, void* value,
+    size_t* sizeRet) {
+  if (session == nullptr || operation == nullptr ||
+      (sites != nullptr && numSites == 0) ||
+      (params != nullptr && numParams == 0) ||
+      (value != nullptr && size == 0) || prop >= QDMI_OPERATION_PROPERTY_MAX) {
+    return QDMI_ERROR_INVALIDARGUMENT;
+  }
+  ADD_STRING_PROPERTY(QDMI_OPERATION_PROPERTY_NAME, operation->name.c_str(),
+                      prop, size, value, sizeRet)
+  ADD_SINGLE_VALUE_PROPERTY(QDMI_OPERATION_PROPERTY_PARAMETERSNUM, size_t,
+                            operation->numParameters, prop, size, value,
+                            sizeRet)
+  if (operation->type != OperationType::SHUTTLING_MOVE) {
+    ADD_SINGLE_VALUE_PROPERTY(QDMI_OPERATION_PROPERTY_DURATION, double,
+                              operation->duration, prop, size, value, sizeRet)
+    ADD_SINGLE_VALUE_PROPERTY(QDMI_OPERATION_PROPERTY_FIDELITY, double,
+                              operation->fidelity, prop, size, value, sizeRet)
+  } else {
+    ADD_SINGLE_VALUE_PROPERTY(QDMI_OPERATION_PROPERTY_DURATION, double, 0.0,
+                              prop, size, value, sizeRet)
+    ADD_SINGLE_VALUE_PROPERTY(QDMI_OPERATION_PROPERTY_FIDELITY, double, 1.0,
+                              prop, size, value, sizeRet)
```

Comment:
Does a shuttling move really always have duration 0 and fidelity 1?
In practice the duration as well as the fidelity depends on the distance of the move, right?

Reply:
Yeah. I suppose returning `NOTSUPPORTED` is the right choice here then.
But actually for both of these properties, as both of them depend on the duration and distance of the move.
The idling decoherence time is something that the device would then have to report as well, but I think we have that tracked somewhere already.